MFVNeutralino/plugins/VertexHistos.cc

```diff
@@ -382,10 +428,131 @@ void MFVVertexHistos::analyze(const edm::Event& event, const edm::EventSetup&) {
   const int nsv = int(auxes->size());
   h_nsv->Fill(nsv, w);
 
+  for (int igenv = 0; igenv < 2; ++igenv) {
+    double genx = mevent->gen_lsp_decay[igenv*3+0];
+    double geny = mevent->gen_lsp_decay[igenv*3+1];
+    double genz = mevent->gen_lsp_decay[igenv*3+2];
+    double genbs2ddist = mevent->mag(genx - mevent->bsx_at_z(genz),
+                                     geny - mevent->bsy_at_z(genz) 
+        );
+    h_genbs2ddist->Fill(genbs2ddist, w);
+  }
+
+  // matching jets with gen quarks from LLPs
+  //double decay_quarks0_pt[2] = {-1,-1};
+  //double decay_quarks1_pt[2] = {-1,-1};
+  //double decaylsp0_pt = -1;
+  //double decaylsp1_pt = -1;
+  //double decay_jets0_pt[2] = {-1,-1};
+  //double decay_jets1_pt[2] = {-1,-1};
+  double nmatched_0 = 0;
+  double nmatched_1 = 0;
+  for (size_t i=0; i<mevent->gen_daughters.size(); ++i){
+    if (abs(mevent->gen_daughter_id[i])==1000022){
+      continue;
+      //if (decaylsp0_pt>0){
+      //  decaylsp1_pt = mevent->gen_daughters[i].Pt();
+      //}
+      //else{
+      //  decaylsp0_pt = mevent->gen_daughters[i].Pt();
+      //}
+    }
+    else{
+      double gd_eta = mevent->gen_daughters[i].Eta();
+      double gd_phi = mevent->gen_daughters[i].Phi();
+      //double dR2_min = 999;
+      int n_matched = 0;
+      //double pt_sum = 0;
+      for (int ij = 0; ij<mevent->njets(); ++ij){
+        double dR2 = reco::deltaR2(mevent->nth_jet_eta(ij), mevent->nth_jet_phi(ij), gd_eta, gd_phi);
+        //double dR2 = (mevent->nth_jet_eta(ij)-gd_eta)*(mevent->nth_jet_eta(ij)-gd_eta)+(mevent->nth_jet_phi(ij)-gd_phi)*(mevent->nth_jet_phi(ij)-gd_phi);
+        //if (dR2_min>dR2){
+        //  dR2_min = dR2;
+        //}
+        if (dR2<0.16){
+          n_matched += 1;
+          //pt_sum += mevent->nth_jet_pt(ij);
+        }
+      }
+      if (i<3){
+        nmatched_0 += n_matched;
+        //if (decay_quarks0_pt[0]>=0){
+        //  decay_quarks0_pt[1] = mevent->gen_daughters[i].Pt();
+        //  decay_jets0_pt[1] = pt_sum;
+        //}
+        //else{
+        //  decay_quarks0_pt[0] = mevent->gen_daughters[i].Pt();
+        //  decay_jets0_pt[0] = pt_sum;
+        //}
+      }
+      else{
+        nmatched_1 += n_matched;
+        //if (decay_quarks1_pt[0]>=0){
+        //  decay_quarks1_pt[1] = mevent->gen_daughters[i].Pt();
+        //  decay_jets1_pt[1] = pt_sum;
+        //}
+        //else{
+        //  decay_quarks1_pt[0] = mevent->gen_daughters[i].Pt();
+        //  decay_jets1_pt[0] = pt_sum;
+        //}
+      }
+    }
+  }
+  h_sv_nsv_nmatchjet->Fill(nmatched_0+nmatched_1, nsv, w);
+
```

Comment:
I think this is similar to elsewhere, that it's either not used anymore or just could do with some comments.